### PR TITLE
feat [ci.jenkins.io, trusted.ci.jenkins.io] allow Azure VM Agent to use private IPs

### PR DESCRIPTION
**Pull request by dduportal**
This PR introduces the ability to use private IPs for Azure Agent.

It also fixes the actual errors on the test harness and vagrant VM (that was also breaking the CI builds) by fixing a nil pointer in the template for Casc clouds.

### CODE REVIEW

**Review by timja — DISMISSED** [review]

**Review by timja — APPROVED** [review]

File: hieradata/clients/azure.ci.jenkins.io.yaml [hunk 1]
```diff
@@ -178,6 +178,10 @@ profile::buildmaster::cloud_agents:
         maxInstances: 10
         useAsMuchAsPosible: true
         credentialsId: "jenkinsvmagents-userpass"
+        privateIP: true
```
Comment (timja) [inline hieradata/clients/azure.ci.jenkins.io.yaml:181]: this should be `usePrivateIP`, applies in a few places

Comment (dduportal) [reply hieradata/clients/azure.ci.jenkins.io.yaml:181]: https://github.com/jenkins-infra/jenkins-infra/pull/1928 Thanks TIm!
